lib/Net/Kubernetes/Resource/Deployment.pm

```diff
@@ -0,0 +1,27 @@
+package Net::Kubernetes::Resource::Deployment;
+
+use Moose;
+
+extends 'Net::Kubernetes::Resource';
+with 'Net::Kubernetes::Resource::Role::State';
+with 'Net::Kubernetes::Resource::Role::Spec';
+with 'Net::Kubernetes::Resource::Role::HasPods';
+with 'Net::Kubernetes::Resource::Role::HasReplicaSets';
+
+sub scale {
+    my($self, $replicas, $timeout) = @_;
+    $timeout ||= 5;
+    $self->spec->{replicas} = $replicas;
+    $self->update;
+    my $st = time;
+    while((time - $st) < $timeout){
+        my $pods = $self->get_pods;
+        if(scalar(@$pods) == $replicas){
+            return "scaled";
```

Comment:
Yeah, I agree. I was trying to keep this consistent with the scale method in replication controllers. In fact, it's really the same method. Could extract that out to yet another role.